lib/modules/datasource/deno/index.ts

```diff
@@ -0,0 +1,117 @@
+import is from '@sindresorhus/is';
+import pMap from 'p-map';
+import { logger } from '../../../logger';
+import { cache } from '../../../util/cache/package/decorator';
+import { regEx } from '../../../util/regex';
+import { joinUrlParts } from '../../../util/url';
+import * as semanticVersioning from '../../versioning/semver';
+import { Datasource } from '../datasource';
+import type { Release } from '../index';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import type {
+  DenoAPIModuleResponse,
+  DenoAPIModuleVersionResponse,
+} from './types';
+import { createSourceURL, tagsToRecord } from './utils';
+
+export class DenoDatasource extends Datasource {
+  static readonly id = 'deno';
+
+  override readonly customRegistrySupport = true;
+
+  override readonly registryStrategy = 'first';
+
+  override readonly defaultVersioning = semanticVersioning.id;
+
+  override readonly defaultRegistryUrls = ['https://apiland.deno.dev'];
+
+  constructor() {
+    super(DenoDatasource.id);
+  }
+
+  @cache({
+    namespace: `datasource-${DenoDatasource.id}`,
+    key: ({ packageName, registryUrl }: GetReleasesConfig) =>
+      // TODO: types (#7154)
+      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+      `${registryUrl}:${packageName}`,
+  })
+  async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    const massagedRegistryUrl = registryUrl ?? this.defaultRegistryUrls[0];
+
+    const extractResult = regEx(
+      '^(https:\\/\\/deno.land\\/)(?<rawPackageName>[^@\\s]+)'
+    ).exec(packageName);
+    if (is.nullOrUndefined(extractResult)) {
```

Comment:
combine with next condition. add `?.groups` to previous exec, so this condition will be much easier

Reply:
My idea has been to give a more clear error message, but I will change it.

Reply:
you can't distinguish those cases, as the group is always there if the regex matches, as it's not optional. So makes no sense to try to create two conditions. it's just typescript, which doesn't know about that fact.